avm/src/vm/opcode.rs

```diff
@@ -3019,8 +3019,8 @@ pub static OPCODES: Lazy<HashMap<i64, ByteOpcode>> = Lazy::new(|| {
       std::process::exit(3);
     }
     // Await HTTP response from the user code
-    let response_hm = match rx.changed().await {
-      Ok(_) => rx.borrow(),
+    let response_hm = match rx.await {
+      Ok(hm) => hm,
       Err(_) => panic!("Failed to receive a response for an HTTP request"),
     };
```

Comment:
Can the `rx` channel decide to close itself out? We may want to do something other than a `panic!` in the error path if that is the case.

Reply:
in an offline conversation we said we would return an error response (I added a TODO for that right after you left this comment).